Sources/Basic/OutputByteStream.swift

```diff
@@ -405,3 +405,95 @@ public struct Format {
         return TransformedSeparatedListStreamable(items: items, transform: transform, separator: separator)
     }
 }
+
+/// Represents a stream which is backed to a file. Not for instantiating.
+public class FileOutputByteStream: OutputByteStream {
+
+    /// Closes the file flushing any buffered data.
+    public func close() {
+        // For subclasses to implement.
+    }
+}
+
+import libc
+
+/// Implements file output stream for local file system.
+public final class LocalFileOutputByteStream: FileOutputByteStream {
+    
+    /// The pointer to the file.
+    let fp: UnsafeMutablePointer<FILE>
+
+    /// Instantiate using the file pointer.
+    init(fp: UnsafeMutablePointer<FILE>) throws {
+        self.fp = fp
+        super.init()
+    }
+
+    /// Opens the file for writing at the provided path.
+    public init(_ path: AbsolutePath) throws {
+        guard let fp = fopen(path.asString, "wb") else {
+            throw FileSystemError(errno: errno)
+        }
+        self.fp = fp
+        super.init()
+    }
+    
+    override public func flush() throws {
+        var contents = buffer
+        while true {
+            let n = fwrite(&contents, 1, contents.count, fp)
+            if n < 0 {
+                if errno == EINTR { continue }
+                throw FileSystemError.ioError
+            }
+            if n != contents.count {
+                throw FileSystemError.ioError
+            }
+            break
+        }
+        // Flush to file too.
+        fflush(fp)
+        // Empty the buffer because we have already written to file.
+        buffer = []
+    }
+
+    override public func close() {
+        // Flush if possible and close the file.
+        try? flush()
+        fclose(fp)
+    }
+}
+
+/// Public stdout stream instance.
+public var stdoutStream: FileOutputByteStream = try! LocalFileOutputByteStream(fp: libc.stdout)
+
+/// Public stderr stream instance.
+public var stderrStream: FileOutputByteStream = try! LocalFileOutputByteStream(fp: libc.stderr)
+
+/// Implements file output stream for in-memory file system.
+public final class InMemoryFileOutputByteStream: FileOutputByteStream {
```

Comment:
I think this implementation should be private to `FileSystem`, not part of `OutputByteStream` per se.